static/app/components/idBadge/teamBadge/index.tsx

```diff
@@ -1,57 +1,40 @@
 import * as React from 'react';
-import isEqual from 'lodash/isEqual';
 
 import TeamStore from 'sentry/stores/teamStore';
 import {Team} from 'sentry/types';
 
-import Badge from './badge';
+import Badge, {BadgeProps} from './badge';
 
-type Props = React.ComponentProps<typeof Badge>;
+function TeamBadge(props: BadgeProps) {
+  const [team, setTeam] = React.useState<Team>(props.team);
 
-type State = {
-  team: Team;
-};
+  const onTeamStoreUpdate = React.useCallback(
```

Comment:
Do you need useCallback? It looks like this function could be inlined or defined in the `useEffect` handler.

Reply:
Yeah, not necessary, I did so to keep the definition decoupled from the listener